src/GraphQL.MicrosoftDI/ConnectionResolverBuilder.cs

```diff
@@ -0,0 +1,405 @@
+        /// Specifies a type that is to be resolved via dependency injection during the resolver's execution.
+        /// </summary>
+        public ConnectionResolverBuilder<TSourceType, TReturnType, T1> WithService<T1>()
+            => new ConnectionResolverBuilder<TSourceType, TReturnType, T1>(_builder, _scoped);
+
+        /// <summary>
+        /// Specifies types that are to be resolved via dependency injection during the resolver's execution.
+        /// </summary>
+        public ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2> WithServices<T1, T2>()
+            => new ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2>(_builder, _scoped);
+
+        /// <inheritdoc cref="WithServices{T1, T2}"/>
+        public ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3> WithServices<T1, T2, T3>()
+            => new ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3>(_builder, _scoped);
+
+        /// <inheritdoc cref="WithServices{T1, T2}"/>
+        public ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4> WithServices<T1, T2, T3, T4>()
+            => new ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4>(_builder, _scoped);
+
+        /// <inheritdoc cref="WithServices{T1, T2}"/>
+        public ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5> WithServices<T1, T2, T3, T4, T5>()
+            => new ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5>(_builder, _scoped);
+
+        /// <summary>
+        /// Specifies that the resolver should run within its own dependency injection scope.
+        /// </summary>
+        public ConnectionResolverBuilder<TSourceType, TReturnType> WithScope()
+        {
+            _scoped = true;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies the return data type of the connection resolver. Typically this would be <see cref="Connection{TNode}"/>
+        /// or <see cref="Connection{TNode, TEdge}"/>, but it can be any type that returns the proper fields for the
+        /// connection graph type.
+        /// <br/><br/>
+        /// For instance, to specify a return type for a Widget you might write:
+        /// <br/><br/>
+        /// <code>
+        /// .Returns&lt;Connection&lt;Widget&gt;&gt;()
+        /// </code>
+        /// </summary>
+        public ConnectionResolverBuilder<TSourceType, TNewReturnType> Returns<TNewReturnType>()
+            => new ConnectionResolverBuilder<TSourceType, TNewReturnType>(_builder, _scoped);
+
+        /// <summary>
+        /// Specifies the delegate to execute when the field is being resolved.
+        /// </summary>
+        public void Resolve(Func<IResolveConnectionContext<TSourceType>, TReturnType> resolver)
+        {
+            if (_scoped)
+                _builder.ResolveScoped(resolver);
+            else
+                _builder.Resolve(context => resolver(context));
+        }
```

Comment:
```suggestion
        => _scoped ? _builder.ResolveScoped(resolver) : _builder.Resolve(resolver);
```

Reply:
See other notes on this.  Same issues are present, where the return value of `.Resolve()` is `object` rather than `T`, so a lambda is necessary.

Note that if `.Returns<TNewReturnType>()` was removed from the builder, it would always accept an `object` and then it could be passed directly through.  I favored a typed context.  We could move the `.Returns<TNewReturnType>()` method to the connection builder in GraphQL.NET (to match the field builder's `.Return<TNewReturnType>()` already present there), but this will change a bunch of APIs in the main repo.  Not sure if they would be breaking or not; I can look into this now if you want me to, or we can just mark this as to-do for v5.

Reply:
> We could move the .Returns<TNewReturnType>() method to the connection builder in GraphQL.NET (to match the field builder's .Return<TNewReturnType>() already present there)

I would have done so.

Reply:
With #2458 this is now able to be completed and has been done.